cql3/query_processor.cc

```diff
@@ -55,6 +55,10 @@ struct query_processor::remote {
     seastar::gate gate;
 };
 
+bool query_processor::topology_global_queue_empty() {
```

Comment:
> so when ALTER KS is executed, we'll have to check if any other global topo req is ongoing and fail the req if that's the case.

Is it acceptable? For instance "cleanup" and "alter KS" may be ran by different people and the former may take a lot of time.

Reply:
cleanup is not a good example since it involves vnodes and so it doesn't conflict with rf change on tablets keyspace. But tablet related transitions would conflict with changing the rf.

Can we check the queue per keyspace?

Reply:
> cleanup is not a good example since it involves vnodes 

Why does it matter what it involves? It will cause alter ks to fail.

Reply:
> > cleanup is not a good example since it involves vnodes
> 
> Why does it matter what it involves? It will cause alter ks to fail.

It's simpler to first restrict the conflicts by scope (keyspace) rather than by type.
Then, we can consider higher grain resolution by type within the keyspace.
Right now it would be acceptable to fail rf change if there's any other admin operation on the keyspace, like another rf-change, or even repair.  As for tablet migration, we should serialize them with rf changes, but alter ks shouldn't fail if there are migrations going on in the background, but rather it should serialize with them.
It is questionable whether rf change should fail if there is a potentially long running node operation ongoing, like decommission or remove node.  If we can detect them I think it would be better to fail rf changes until they're done.

(Tangential to this, since we can't decommission or remove a node if we don't have enough remaining nodes to satisfy the replication factor, it would be nice to have a way to reduce the replication factor by specifying a hint to the particular node we want to evict when reducing RF)

Reply:
@gleb-cloudius in any case, I think it would be acceptable in 6.0.0 for global cleanup and rf change to conflict, and we can consider improving/fixing that as a follow up.

Reply:
> @gleb-cloudius in any case, I think it would be acceptable in 6.0.0 for global cleanup and rf change to conflict, and we can consider improving/fixing that as a follow up.

OK. I am just pointing out the difference between existing cases where only admin commands may fail and here , where DDL may conflict with an admin command. If this is acceptable for 6.0 so be it.

Reply:
OK, this point will be considered when planning follow-ups of this PR.

Reply:
Created issue - https://github.com/scylladb/scylladb/issues/19127